network-api/networkapi/wagtailpages/migrations/0142_alter_blogindexpage_callout_box.py

```diff
@@ -0,0 +1,340 @@
+# Generated by Django 4.2.11 on 2024-06-23 22:18
+
+import wagtail.blocks
+import wagtail.documents.blocks
+import wagtail.fields
+import wagtail.snippets.blocks
+import wagtailmedia.blocks
+from django.db import migrations
+
+import networkapi.wagtailpages.pagemodels.blog.blog_topic
+import networkapi.wagtailpages.validators
+
+
+def migrate_blogindexpage_callout_box(apps, schema_editor):
+    BlogIndexPage = apps.get_model("wagtailpages", "BlogIndexPage")
+
+    for page in BlogIndexPage.objects.all():
+        if not page.callout_box:
+            continue
+
+        new_callout_box_data = []
+
+        for block in page.callout_box:
+            if block.block_type == "callout_box":
+                new_value = {
+                    "title": block.value.get("title"),
+                    "related_topics": block.value.get("related_topics"),
+                    "show_icon": block.value.get("show_icon"),
+                    "body": block.value.get("body"),
+                    "audio": block.value.get("audio"),
+                    "link_button": [],
+                }
+                if block.value.get("link_button_text") and block.value.get("link_button_url"):
+                    new_value["link_button"].append(
+                        {
+                            "label": block.value.get("link_button_text"),
+                            "link_to": "external_url",
+                            "external_url": block.value.get("link_button_url"),
+                            "new_window": True,
+                        }
+                    )
+                new_callout_box_data.append(("callout_box", new_value))
+
+        page.callout_box = new_callout_box_data
+        page.save()
```

Comment:
Here is the updated migration operation, this operation loops through every `BlogIndexPage` in the DB, (which should be 9, 1 for each translation of https://foundation.mozilla.org/blog), and then updates the existing link data to the new `LinkBlock` format 👍